### PR TITLE
Fixed a bug in the element deactivation approach

### PR DESCRIPTION
**Pull request by wenqing**
The bug: `DirichletBoundaryConditionWithinTimeInterval::_nodes_in_bc_mesh` stored the bulk nodes not the nodes of BC mesh before  this correction. Now it stores the nodes from BC mesh.  
(**Comment**: it would be better to avoid copying nodes and elements to sub meshes for BC from the bulk mesh )  

1. [ ] Feature description was added to the [changelog](https://github.com/ufz/ogs/wiki/Release-notes-6.3.0)
2. [x] A 3D Test covering the bug fixing was added.
3. [ ] Any new feature or behavior change was documented? TODO in another PR.

element_deactivation_3D

### CODE REVIEW

**Comment by wenqing** [comment]
Commit 3253bd0  [Changed the search method] just changed the search method to find the inactive nodes. The previous implementation of  searching  inactive nodes work fines but with an additional memory of an array, the present one removes the additional array with a tiny increment in computation.

File: ProcessLib/DeactivatedSubdomain.cpp [hunk 1]
```diff
@@ -89,49 +89,39 @@ std::unique_ptr<DeactivatedSubdomain const> createDeactivatedSubdomain(
...
+            if (std::find_if(
+                    connected_elements.begin(),
+                    connected_elements.end(),
+                    [&](auto const* const connected_elem) -> bool {
+                        return ids != (*material_ids)[connected_elem->getID()];
+                    }) != connected_elements.end())
+            {
+                continue;
+            }
+            else
```
Comment (endJunction) [inline ProcessLib/DeactivatedSubdomain.cpp]: no need for an `else` branch after `return`, `continue`, `break`...

File: ProcessLib/DeactivatedSubdomain.cpp [hunk 2]
```diff
@@ -89,55 +88,78 @@ std::unique_ptr<DeactivatedSubdomain const> createDeactivatedSubdomain(
...
+            {
+                continue;
+            }
+            deactivated_nodes.push_back(
+                const_cast<MeshLib::Node*>(nodes_in_bc_mesh[i]));
+
+            deactivated_bulk_node_ids.erase(
+                deactivated_bulk_node_ids.begin() +
+                std::distance(deactivated_bulk_node_ids.begin(),
+                              found_iterator));
```
Comment (endJunction) [inline ProcessLib/DeactivatedSubdomain.cpp]: Isn't it same as
```c++
deactivated_bulk_node_ids.erase(found_iterator);
```
?

File: ProcessLib/DeactivatedSubdomain.cpp (hunk 1 above)
Comment (wenqing) [reply ProcessLib/DeactivatedSubdomain.cpp]: Forgot to remove that after removed lines except for `continue;` in if block. Removed the else by amending the commit.

File: ProcessLib/DeactivatedSubdomain.cpp (hunk 2 above)
Comment (wenqing) [reply ProcessLib/DeactivatedSubdomain.cpp]: You are right. Changed by amending the commit.

**Pull request merged**